File: tests/cleanup.test.ts
import * as sh from 'shelljs'
import test from 'ava'

const workloadIdentityNamespace = "azure-workload-identity-system"
const RUN_WORKLOAD_IDENTITY_TESTS = process.env['AZURE_RUN_WORKLOAD_IDENTITY_TESTS']

test.before('setup shelljs', () => {
  sh.config.silent = true
})

test.serial('Remove KEDA', t => {
  let result = sh.exec('(cd .. && make undeploy)')
  if (result.code !== 0) {
    t.fail('error removing keda. ' + result)
  }
  t.pass('KEDA undeployed successfully using make undeploy command')
})

test.serial('remove azure workload identity kubernetes components', t => {
  if (!RUN_WORKLOAD_IDENTITY_TESTS || RUN_WORKLOAD_IDENTITY_TESTS == 'false') {
    t.pass('skipping as workload identity tests are disabled')
    return
  }

  t.is(0,
    sh.exec(`helm uninstall workload-identity-webhook --namespace ${workloadIdentityNamespace}`).code,
    'should be able to uninstall workload identity webhook'
  )

  sh.exec(`kubectl delete ns ${workloadIdentityNamespace}`)
})

File: tests/cleanup_test.go
//go:build e2e
// +build e2e

package tests

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/require"

	. "github.com/kedacore/keda/v2/tests"
)

func TestRemoveKEDA(t *testing.T) {
	out, err := ExecuteCommandWithDir("make undeploy", "..")
	require.NoErrorf(t, err, "error removing KEDA - %s", err)

	t.Log(string(out))
	t.Log("KEDA removed successfully using 'make undeploy' command")
}

func TestRemoveWorkloadIdentityComponents(t *testing.T) {
	if AzureRunWorkloadIdentityTests == "" || AzureRunWorkloadIdentityTests == "false" {
		t.Skip("skipping as workload identity tests are disabled")
	}

	_, err := ExecuteCommand(fmt.Sprintf("helm uninstall workload-identity-webhook --namespace %s", AzureWorkloadIdentityNamespace))
	require.NoErrorf(t, err, "cannot uninstall workload identity webhook - %s", err)

	Kc = GetKubernetesClient(t)

	DeleteNamespace(t, Kc, AzureWorkloadIdentityNamespace)
}
